xmodule/capa/capa_problem.py

```diff
@@ -467,12 +480,13 @@ def get_grade_from_current_answers(self, student_answers):
 
         For rescoring, `student_answers` is None.
 
+        If `correct_map` is provided, it is used as the correct_map for grading.
+
         Calls the Response for each question in this problem, to do the actual grading.
         """
-        # old CorrectMap
-        oldcmap = self.correct_map
+        if not correct_map:
+            correct_map = self.correct_map
```

Comment:
May I ask why we are adding this here?

Reply:
This method now optionally accepts a `correct_map` parameter. If not passed, it will work as it has always worked, taking the `correct_map` from the instance (`self.correct_map`).

The new parameter was added so that this method can be used for the modified rescoring functionality, which is now performed for each of the attempts made, and not just the last one.

Reply:
It makes sense for backwards compatibility reasons not to change the signature for `get_grade_from_current_answers()`, yet the two methods (`get_grade_from_current_answers()` and `get_grade_from_answers()` are sufficiently similar that the current implementation seems to run afoul of the Do Not repeat Yourself (DRY) principle. I know I had to spend time and careful attention to convince myself that I understood where the differences lie.

Could you please reimplement `get_grade_from_current_answers()` so that it merely delegates to `get_grade_from_answers()`, passing the instance's `correct_map` dictionary as an argument? That will remove the code duplication without hurting backwards compatibility.